systemvm/debian/etc/cron.hourly/clear_cache

```diff
@@ -0,0 +1,8 @@
+#!/bin/bash
+
+# clear memory cache to ultimately reduce swapping
+
+phymem=$(free|awk '/^Mem:/{print $2}')
+if [ $phymem -lt 513000 ]; then
+  sync && echo 1 > /proc/sys/vm/drop_caches
```

Comment:
Same applies here. The kernel is the one who clears the page cache if it needs the memory for something else

Reply:
As recommended amount of memory for a Debian host is 4GB as we are working **so** far below that, I would say that we are operating well outside 'normal' practices.  Personally, I have long questioned the use of Debian, and would have thought FreeBSD was a better choice.  wrt to memory, the minimum memory for Debian is 96MB.

at 256MB RAM we can't really afford almost ANY caching of disk to memory.  If we started at 4GB then i wouldn't dream of touching this.

The specific, highly replicable issue that is designed to fix, is that the cached disk access is not given back, instead the host starts swapping memory to disk, quickly making the VR unusably due to the CPU usage that this causes (see  https://github.com/apache/cloudstack/issues/3039)

I hope you can see why this sub-optimal fix is proposed to remedy the massively sub-optimal host spec that the VR is trying to operate in.

This *ridiculous* level of outside intervention is only being applied to VRs with 512MB or less available to it, which itself is a ridiculously low amount of RAM to give a Debian host.

I should add that this was 'discovered' and tested through manually monitoring the VR memory usage over time and determining that the disk cache was the 'culprit' for the growing memory footprint.